File: app/src/main/java/com/carrati/lebooks/Database/IMyBooksDAO.java
package com.carrati.lebooks.Database;

import com.carrati.lebooks.Model.Book;

import java.util.List;

public interface IMyBooksDAO {
    boolean salvar(Book book);
    boolean procurarLivro(Book book);
    List<Book> listar();
    void cleanBooks();
}

File: app/src/main/java/com/carrati/lebooks/Database/IMyBooksDAO.kt
package com.carrati.lebooks.Database

import com.carrati.lebooks.Model.Book

interface IMyBooksDAO {
    fun salvarLivroComprado(book: Book): Boolean
    fun procurarLivroComprado(book: Book): Boolean
    fun listarLivroComprado(): List<Book>
    fun salvarLivroFavorito(book: Book): Boolean
    fun procurarLivroFavorito(book: Book): Boolean
    fun deletarLivroFavorito(book: Book): Boolean
    fun cleanAllBooks()
}
